### ISSUE TITLE
swift-nio-http2 causing conflicting requirements

### ISSUE THREAD

**Issue by AaronHarden** [grpc/grpc-swift#400]
```
swift build -Xcc -ISources/BoringSSL/include -Xlinker -lz --product protoc-gen-swift -c release -Xswiftc -static-stdlib
Updating https://github.com/apple/swift-protobuf.git
Updating https://github.com/apple/swift-nio-zlib-support.git
Updating https://github.com/apple/swift-nio-nghttp2-support.git
Updating https://github.com/kylef/Commander.git
Updating https://github.com/kylef/Spectre.git
Updating https://github.com/apple/swift-nio.git
Updating https://github.com/apple/swift-nio-http2.git
error: dependency graph is unresolvable; found these conflicting requirements:

Dependencies: 
    https://github.com/apple/swift-nio-http2.git @ 0.2.1..<0.3.0
error: product dependency 'SwiftProtobuf' not found
error: product dependency 'NIOFoundationCompat' not found
error: product dependency 'NIOHTTP1' not found
error: product dependency 'NIOHTTP2' not found
error: product dependency 'SwiftProtobuf' not found
error: product dependency 'SwiftProtobuf' not found
error: product dependency 'Commander' not found
error: product dependency 'SwiftProtobuf' not found
error: product dependency 'SwiftProtobufPluginLibrary' not found
error: product dependency 'protoc-gen-swift' not found
error: product dependency 'SwiftProtobuf' not found
error: product dependency 'Commander' not found
error: product dependency 'Commander' not found
make: *** [plugin] Error 1```


What would cause SPM to fail to find it?

**Comment by MrMage** [comment]
What does your `Package.swift` look like? Did you try deleting your entire `.build` folder to start over?

**Comment by weissi** [comment]
this might be due to https://bugs.swift.org/browse/SR-6392 . Unfortunately, SwiftPM's current error messages are often pretty useless, AFAIK it can be pretty much any package in the dependency tree that it prints as a problem. Fortunately, help's on the way: https://github.com/apple/swift-package-manager/pulls?q=is%3Apr+pubgrub+is%3Aclosed

**Comment by AaronHarden** [comment]
> What does your `Package.swift` look like? Did you try deleting your entire `.build` folder to start over?

Fresh clone from master. Using Xcode 9.2, would downgrading to 9.1 have any effect?

**Comment by MrMage** [comment]
> Fresh clone from master. Using Xcode 9.2, would downgrading to 9.1 have any effect?

Hmm, could you try upgrading to Xcode 10.1 instead?

**Comment by AaronHarden** [comment]
Unfortunately stuck using Sierra so can't install 10.1. Already tried installing nghttp2 with no change.

**Comment by weissi** [comment]
@AaronHarden okay, `swift-nio-http2` only supports Swift 4.1+. Swift 4.1 shipped with Xcode 9.3 so that's the minimum version you will need.
